packages/eas-cli/src/commands/project/init.ts

```diff
@@ -1,27 +1,226 @@
+import { Flags } from '@oclif/core';
 import chalk from 'chalk';
+import nullthrows from 'nullthrows';
+import terminalLink from 'terminal-link';
 
+import { getProjectDashboardUrl } from '../../build/utils/url';
 import EasCommand from '../../commandUtils/EasCommand';
+import { AppPrivacy, Role } from '../../graphql/generated';
+import { AppMutation } from '../../graphql/mutations/AppMutation';
 import Log from '../../log';
+import { ora } from '../../ora';
 import { getExpoConfig } from '../../project/expoConfig';
-import { findProjectRootAsync, getProjectIdAsync } from '../../project/projectUtils';
+import { findProjectIdByAccountNameAndSlugNullableAsync } from '../../project/fetchOrCreateProjectIDForWriteToConfigWithConfirmationAsync';
+import {
+  findProjectRootAsync,
+  saveProjectIdToAppConfigAsync,
+  toAppPrivacy,
+} from '../../project/projectUtils';
+import { confirmAsync, promptAsync } from '../../prompts';
+import { findAccountByName } from '../../user/Account';
+import { ensureLoggedInAsync } from '../../user/actions';
+
+type InitializeMethodOptions = {
+  force: boolean;
+  nonInteractive: boolean;
+};
 
 export default class ProjectInit extends EasCommand {
   static override description = 'create or link an EAS project';
   static override aliases = ['init'];
 
-  async runAsync(): Promise<void> {
+  static override flags = {
+    id: Flags.string({
+      description: 'ID of the EAS project to link',
+    }),
+    force: Flags.boolean({
+      description: 'Whether to overwrite any existing project ID',
+      dependsOn: ['id'],
```

Comment:
why require an id? I think it would be useful to resolve id based on the owner field. Dominik suggested in one of the previous PRs to add envs that skip projected requirements, but using `eas init --force && eas build ...` would solve that issue.

e.g.  https://github.com/expo/turtle-v2-example/blob/master/managed/app.config.ts#L9 We need sth like this to be able to build the same project for different accounts and environments, but with --force flag we wouldn't need to define those ids locally.
Also, our current solution has the problem, where the project is not defined during the build, but that is not a problem for us, because we never run that project, just build for testing.

Reply:
Explicit IDs is the direction we're moving in and where we want to end up (deprecating owner and slug fields entirely). Put another way, if we were building EAS from scratch, knowing the requirements we know now we wouldn't add the owner field or the slug field at all. Instead, everything would just be ID based. While it's not as beautiful, it's far more robust, less prone to error, and requires far less engineering attention.

Inferring the account from the owner field in this PR should be considered a fallback and legacy. In the near future, we'll most likely just tell people to run `eas init --id <project_id>` instead of adding an owner field.

For the differing project ID per environment, I think a map in a dynamic config is the most correct and robust approach. There's no question what project this is referring to, and no chance for error if and env is assumed but incorrectly applied. (for example, if some the env is forgotten on a sub-command it fails on the server rather than accidentally apply to a different project implicitly).

Reply:
Sure, that is fair. Our internal use case is a bit of hack on it's own and I see why we would not want to support it.